### ISSUE TITLE
[KCModule: msm_kernel] - problems when downloading packages

### ISSUE THREAD

**Issue by scachemaille** [manjaro/manjaro-settings-manager#60]
There is a problem when the kernel installation take too much time (ex: having do download packages)
the action dialog say that the job is done, but it's not really finished, pacman is still running in background. 

![screenshot_20160427_154741](https://cloud.githubusercontent.com/assets/16231972/14854628/b9433f8a-0c90-11e6-9df3-f9f0a49ef111.png)

I investigated. and I don't understand the problem.
in
`ActionDialog::startJob()`
the function
`if ( job->exec() )` who start the KAuth action
 return false before the pacman process exit in

> ActionReply
> KernelAuthHelper::runPacman( const QVariantMap& args )
> {
>     .....
>     .....
>     ...
>     pacman->waitForFinished( -1 );
>     return actionReply;
> 
> }

Here my debug output (i added qDebug outputs)

```
"16:15:04"  before 'if ( job->exec() )'
Debug message from helper: "16:15:10"  before 'pacman->start( "/usr/bin/pacman", args["arguments"].toStringList() );'
"16:15:29"  after 'if ( job->exec() )'
Debug message from helper: "16:15:50"  after 'pacman->waitForFinished( -1 );'
```

we can see that job-exec() returned 20s before pacman QProcess finished
maybe someone will understand why. I don't.

best regards

**Comment by Kirek** [comment]
Yes, I know of this issue, some change kde frameworks some time ago make it respect the 25 seconds timeout reply of dbus... even if I was using a async job...

**Comment by scachemaille** [comment]
Is it a reported bug or a limitation of this functionality?

**Comment by scachemaille** [comment]
I did some research.  It didn't seem to be a Kauth change.  It seems it never used any timeout in the dbus proxy helper. 
It seems it's not qtdbus code change as it always used a default -1 value if none is given who correspond to the 25s default in the libdbus from system.
But I saw in Qt 5.6 changelog that they now use the system libdbus library by default instead of something else. It's maybe related.  I need some more research. 
I did not really tested msm to install kernel before. 
Does it was really working without a timeout of 25s with kauth?

**Comment by Kirek** [comment]
It already didn't work with qt 5.5, maybe a kauth job wasn't meant to be long lived to begin with and it was a bug they fixed...
I tried various approaches with kauth (sync, async...) and nothing seems to work, maybe calling using qdbus directly will be better, or even use packagekit.

**Comment by scachemaille** [comment]
Using qdbus directly would solve the timeout issue, as we will be able to set a proper timeout.
but how to achieve nicely the the privilege escalation given by kauth to run the process as root?

**Comment by guinux** [comment]
Would you be interested in using the bus daemon written for Pamac to download and install package? It solves the problem with a start job method and a finish job signal. It also sends progress with signals and integrate authentication with polkit.

**Comment by scachemaille** [comment]
I will give a look at it.

**Comment by scachemaille** [comment]
for information.
I asked a "pull request" to add a timeout property to solve the problem.
let's see how it goes.
[pull request](https://git.reviewboard.kde.org/r/128902/)
[issue I created](https://bugs.kde.org/show_bug.cgi?id=363200) but no feedback from devs

**Comment by scachemaille** [comment]
Good news.
My proposal has been accpeted and merged into kauth repository. it should be available in kauth 5.29 (december 2016). 
[comit](https://cgit.kde.org/kauth.git/commit/?id=99938b0980131202c5d3e2fb1845106c147917d1)
[review](https://git.reviewboard.kde.org/r/128902/)
